src/cliffordsynthesis/ExactStrategy.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+* This file is part of the MQT QMAP library which is released under the MIT license.
+* See file README.md or go to https://www.cda.cit.tum.de/research/ibm_qx_mapping/ for more information.
+*/
+
+#include "cliffordsynthesis/ExactStrategy.hpp"
+
+#include "cliffordsynthesis/TargetMetricHandler.hpp"
+namespace cs {
+    void cs::ExactStrategy::runExactStrategy(std::size_t timesteps, const CouplingMap& reducedCM,
+                                             const QubitSubset& qubitChoice, const Configuration& configuration, CliffordSynthesizer& synthesizer) {
+        switch (configuration.strategy) {
+            case OptimizationStrategy::UseMinimizer:
+                runMaxSat(timesteps, reducedCM, qubitChoice, configuration, synthesizer);
+                break;
+            case OptimizationStrategy::StartLow:
+                runStartLow(timesteps, reducedCM, qubitChoice, configuration, synthesizer);
+                break;
+            case OptimizationStrategy::StartHigh:
+                runStartHigh(timesteps, reducedCM, qubitChoice, configuration, synthesizer);
+                break;
+            case OptimizationStrategy::MinMax:
+                runBinarySearch(timesteps, reducedCM, qubitChoice, configuration, synthesizer);
+                break;
+            default:
+                throw std::runtime_error("Unknown optimization strategy");
+        }
+    }
+    void ExactStrategy::runMaxSat(std::size_t timesteps, const CouplingMap& reducedCM, const QubitSubset& qubitChoice, const Configuration& configuration, CliffordSynthesizer& synthesizer) {
+        DEBUG() << "Running minimizer" << std::endl;
+        Results r = synthesizer.mainOptimization(timesteps, reducedCM, qubitChoice, configuration.targetTableau, configuration.initialTableau,
```

Comment:
```suggestion
        const auto r = synthesizer.mainOptimization(timesteps, reducedCM, qubitChoice, configuration.targetTableau, configuration.initialTableau,
```

Reply:
this does not work without changing updateResults, which does not always get a const

Reply:
Any argument against changing the updateResults signature?

Reply:
Not at all, I'll do that